Review pull request

golpashin opened the pull request:
This is a review pull request? maybe???

golpashin committed:
Add files via upload

mahshidmansouri commented:
Hi Alen, 

I will review your code. 

Mahshid

mahshidmansouri commented:
Hi Alen, 

I reviewed your code. Overall, it was well-written. Here are some suggestions that might help with the readability and modularity of your code as well as your results presentation: 

**Code:** 
1. It would be better if you could convert some parts of your into functions instead to increase the readability and modularity of your code. For instance, it would be better if you could define the part of your code where you are training your network to a "train_DQN" function and call that function later on if you need to. This would be similar to how you have defined "greedy_a" for greedy-epsilon action selection or "exp_replay" to define your experience replay buffer. In addition, you could also move all the other side functions/classes that you are using a lot in your code to a "helpers.py" file and import them inside your main code which will help with adding modularity to your code as well as reducing the length of your code. 

2. I would also suggest that you convert the parts of your code for plotting purposes into plotting functions, and move those to a separate .py file (e.g., plotters.py) and call those inside your main code. Then, you will only have a single line of code for calling each plotting function in your main code. This will help with having a much cleaner and more modular main code and if somebody was interested to see the details of each plotting function, they could refer to the plotting file and dig into it. 

4. I would also suggest you add more comments to explain what each of your functions are doing. Indicate what the input parameters to the function are and what the function returns. This will also help with making your code more readable and understandable. 


**Figures/Results:**
1. Please add labels + title to your ablation study plots. In addition, I guess based on the paper, we need to report an average score per each condition, and not a learning curve plot. You could look into the paper again and see how they got the average score per each condition, and use bar charts instead to compare the results. 

Good luck!
Mahshid

golpashin committed:
Delete test_discreteaction_pendulum.py

golpashin committed:
Add files via upload

golpashin commented:
Hi Mahshid,

Thanks you for your review. I addressed and fixed your comments as follows:
1. I converted more of my code to functions, adding for instance a SGD function.
2. I added a plotters.py function that plots the agent trajectories and their respective learning curve
3. I added comments to each function explaining what each function does. I also added details on input and output of the functions. Additionally, I tried to increase the amount of comments I was leaving for each command, so the code is more clear.
Figures/Results
1. I added labels and titles, and also added the ablation study code. It should now be plotting the average learning curves over a certain number of realizations. I compared my results to the paper's table 3 and they seem to make sense.
Thank you so much for your review again!
